### PR TITLE
 DROOLS-5902: Distinguish builtin and simple custom type in DT input column

### PR DESCRIPTION
**Pull request by jomarko**
For more details see: https://issues.redhat.com/browse/DROOLS-5902

### CODE REVIEW

**Comment by jomarko** [comment]
I will rebase this PR once #1383 is merged

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableEditorDefinitionEnricherTest.java [hunk 1]
```diff
@@ -900,6 +902,24 @@ public void testAddInputClauseRequirement() {
         assertEquals(NUMBER.getName(), inputClause2.typeRef.getLocalPart());
     }
 
+    @Test
+    public void testAddInputClauseRequirement_simpleCustomType() {
```
Comment (jomarko) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableEditorDefinitionEnricherTest.java:906]: On `main`, will the test fail as:
Screenshot_20230113_023817

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableEditorDefinitionEnricher.java [hunk 2]
```diff
@@ -375,7 +375,7 @@ void addInputClauseRequirement(final ItemDefinition itemDefinition,
             inputClauseRequirements.add(
                     new ClauseRequirement(
                             text,
-                            getQName(itemDefinition),
+                            getQNameFromItemDefinitionName(itemDefinition),
```
Comment (yesamer) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableEditorDefinitionEnricher.java]: @jomarko If I understood correctly, it doesn't exist a case where `itemDefinition.getTypeRef()` is used. With this change, `getQNameFromItemDefinitionName` will be always used, right?

Comment (jomarko) [reply packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableEditorDefinitionEnricher.java]: I think I do not understand your question, could you plese try to clarify it ?

**Comment by yesamer** [comment]
@jomarko A test is failing:   `DecisionTableEditorDefinitionEnricherTest.testModelEnrichmentWhenInputDataHasConstraints:342 expected:<date> but was:<null>`

**Comment by jomarko** [comment]
@yesamer thank you for review . Failing tests shows me there are few cases, I need to fix, Will let you know when this is again ready for merge.

**Review by yesamer — APPROVED** [review]

**Pull request merged**